spring-boot/proxy/src/main/java/eu/coatrack/proxy/security/ApiKeyAuthTokenVerifier.java

```diff
@@ -20,166 +20,124 @@
-
-            String urlToSearchForApiKeys = securityUtil.attachGatewayApiKeyToUrl(
-                    adminBaseUrl + adminResourceToSearchForApiKeys + apiKeyValue);
-
-            ResponseEntity<ApiKey> resultOfApiKeySearch = restTemplate.getForEntity(urlToSearchForApiKeys, ApiKey.class);
-
-            if (resultOfApiKeySearch != null) {
-                ApiKey apiKey = resultOfApiKeySearch.getBody();
-                if (apiKey != null) {
-                    log.debug("API key was found by CoatRack admin: " + apiKey);
-
-                    if (Date.valueOf(LocalDate.now()).after(apiKey.getValidUntil())) {
-                        throw new CredentialsExpiredException("Api key is expired");
-                    }
-                    if (apiKey.getDeletedWhen() != null) {
-                        return false;
-                    }
-                } else {
-                    log.debug("API key value could not be found by CoatRack Admin: " + apiKeyValue);
-                    return false;
-                }
-                return true;
-            } else {
-                log.error("Communication with Admin server failed, result is: " + resultOfApiKeySearch);
-                throw new AuthenticationServiceException("Communication with auth server failed");
-            }
-        } catch (HttpClientErrorException e) {
-            if (e.getStatusCode().equals(HttpStatus.NOT_FOUND)) {
-                log.debug("API key value is invalid: " + apiKeyValue);
-                return false;
-            } else {
-                log.error("Error when communicating with auth server", e);
-                throw new AuthenticationServiceException("Communication with auth server failed");
-            }
-        }
+    private Authentication createApiKeyAuthToken(Authentication authentication) {
+        String apiKeyValue = getApiKeyValue(authentication);
+        //TODO this is just a workaround for now: check for fixed API key to allow CoatRack admin access
+        if (isAdminsKey(apiKeyValue)) {
+            return createAdminsAuthToken(apiKeyValue);
+        } else
+            return verifyApiKeyAndIfAuthorizedCreateConsumerAuthToken(apiKeyValue);
     }
 
-    private ServiceApi getServiceApiByApiKey(String apiKeyValue) {
+    private String getApiKeyValue(Authentication authentication) {
+        log.debug("Getting API key value from authentication {}.", authentication.getName());
+        Assert.notNull(authentication.getCredentials());
+        Assert.isInstanceOf(String.class, authentication.getCredentials());
+        String apiKeyValue = (String) authentication.getCredentials();
+        Assert.hasText(apiKeyValue);
+        return apiKeyValue;
+    }
 
-        try {
-            log.debug("trying to get service api entity by APIKEY value {}", apiKeyValue);
+    private boolean isAdminsKey(String apiKeyValue) {
```

Comment:
The trailing `s` looks to me like a "plural s" rather than "genitive s", so I would suggest to remove it, and either use just
```suggestion
    private boolean isAdminKey(String apiKeyValue) {
```
or a more descriptive option like 
```suggestion
    private boolean isApiKeyOfCoatrackAdmin(String apiKeyValue) {
```

Reply:
Done.